### PR TITLE
Move the failing case (sampling integer cubemap textures) to a separa…

### PR DESCRIPTION
**Pull request by zhenyao**
…te tests.

The purpose is to isolate the failing case so we can skip in WebGL 2.0.0 and leave it
for future versions where the driver bug can be fixed or worked around.

https://github.com/KhronosGroup/WebGL/issues/1819

### CODE REVIEW

**Comment by zhenyao** [comment]
@kenrussell Please review.

File: sdk/tests/js/tests/tex-image-and-sub-image-2d-with-webgl-canvas.js [hunk 1]
```diff
@@ -154,14 +165,14 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
...
+            wtu.checkCanvasRect(gl, 0, bottom, (skipCorner && !flipY) ? halfWidth : width, halfHeight, redColor,
                     "shouldBe " + redColor);
             debug("Checking " + (flipY ? "bottom" : "top"));
-            wtu.checkCanvasRect(gl, 0, top, width, halfHeight, greenColor,
+            wtu.checkCanvasRect(gl, 0, top, (skipCorner && flipY) ? halfWidth : width, halfHeight, greenColor,
                     "shouldBe " + greenColor);
         }
 
-        if (false) {
+        if (true) {
```
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-2d-with-webgl-canvas.js]: Is the debugging image still desired? If so, that's fine.

**Comment by kenrussell** [comment]
Excellent work Mo diagnosing this bug and splitting off the failing test cases. LGTM from my standpoint.

One question, but it's minor, and could be addressed later. Merging now.

**Comment by Richard-Yunchao** [comment]
I think we should not skip these tests in WebGL 2.0.1. WDYT? @zhenyao , @kenrussell . We can file radar to Apple if necessary.

FYI. @gyagp

**Comment by kenrussell** [comment]
@Richard-Yunchao @gyagp These tests are enabled in the WebGL 2.0.1 test suite.

The associated Chromium bug is http://crbug.com/658930 and the Radar was 27231650. It looks like it's been fixed on Intel GEN9 GPUs on macOS but not on GEN7. If fixing these issues on GEN7 is a priority then please help us file a new Radar. Thanks.

**Pull request merged**